### ISSUE TITLE
Cleanup imports for build/config/features.gni from buildflags.gni files

### ISSUE THREAD

**Issue by mariospr** [brave/brave-browser#15171]
## Description 

While working on https://github.com/brave/brave-core/pull/8443, and before addressing the review comments from https://github.com/brave/brave-core/pull/8443#pullrequestreview-629335148, I found out that I would have to drop a `import("//build/config/features.gni")` from several `buildflags.gni` files to avoid a dependency cycle.

While this is not needed in that PR anymore since [we're following a different approach now](https://github.com/brave/brave-core/pull/8443#discussion_r608489705), it made me realize that such imports are not strictly needed and therefore it would be a good idea to remove them where possible to avoid potentially running into strange dependency cycles like those ones in the future, so here's an issue to track that work down.

In other words, the **task at hand** would be to **remove the explicit `import("//build/config/features.gni")` likes from Brave's `buildflags.gni` files where possible**, to avoid running into dependency cycles in the future.
